jina/serve/runtimes/gateway/http/fastapi.py

```diff
@@ -0,0 +1,117 @@
+    implementing the `app` property. This property should return a fastapi app. The base Gateway will handle starting
+    a server and serving the application using that server."""
+
+    def __init__(
+        self,
+        ssl_keyfile: Optional[str] = None,
+        ssl_certfile: Optional[str] = None,
+        uvicorn_kwargs: Optional[dict] = None,
+        proxy: bool = False,
+        **kwargs
+    ):
+        """Initialize the FastAPIBaseGateway
+        :param ssl_keyfile: the path to the key file
+        :param ssl_certfile: the path to the certificate file
+        :param uvicorn_kwargs: Dictionary of kwargs arguments that will be passed to Uvicorn server when starting the server
+        :param proxy: If set, respect the http_proxy and https_proxy environment variables, otherwise, it will unset
+            these proxy variables before start. gRPC seems to prefer no proxy
+        :param kwargs: keyword args
+        """
+        super().__init__(**kwargs)
+        self._ssl_keyfile = ssl_keyfile
+        self._ssl_certfile = ssl_certfile
+        self.uvicorn_kwargs = uvicorn_kwargs
+
+        if not proxy and os.name != 'nt':
+            os.unsetenv('http_proxy')
+            os.unsetenv('https_proxy')
+
+    @property
+    @abstractmethod
+    def app(self):
+        '''Get a FastAPI app'''
+        ...
+
+    async def setup_server(self):
+        """
+        Initialize and return GRPC server
+        """
+        with ImportExtensions(required=True):
+            from uvicorn import Config, Server
+
+        class UviServer(Server):
+            """The uvicorn server."""
+
+            async def setup(self, sockets=None):
+                """
+                Setup uvicorn server.
+
+                :param sockets: sockets of server.
+                """
+                config = self.config
+                if not config.loaded:
+                    config.load()
+                self.lifespan = config.lifespan_class(config)
+                await self.startup(sockets=sockets)
+                if self.should_exit:
+                    return
+
+            async def serve(self, **kwargs):
+                """
+                Start the server.
+
+                :param kwargs: keyword arguments
+                """
+                await self.main_loop()
+
+        if 'CICD_JINA_DISABLE_HEALTHCHECK_LOGS' in os.environ:
+
+            class _EndpointFilter(logging.Filter):
+                def filter(self, record: logging.LogRecord) -> bool:
+                    # NOTE: space is important after `GET /`, else all logs will be disabled.
+                    return record.getMessage().find("GET / ") == -1
+
+            # Filter out healthcheck endpoint `GET /`
+            logging.getLogger("uvicorn.access").addFilter(_EndpointFilter())
+
+        uvicorn_kwargs = self.uvicorn_kwargs or {}
+
+        if self._ssl_keyfile and 'ssl_keyfile' not in uvicorn_kwargs.keys():
```

Comment:
I am thinking that it may be useless to have `ssl_keyfile` beyond the `uvicorn_kwargs`?